lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -1435,6 +1461,24 @@ internal class EvaluatingCompiler(
                     // bind the joined value to the bindings for the filter/project
                     FromProduction(joinedValues, fromEnv.nest(localsBinder.bindLocals(joinedValues)))
                 }
+            // Nest LET bindings in the FROM environment
+            if (compiledLetSources != null) {
+                seq = seq.map { fromProduction ->
```

Comment:
Does kotlin `map` process a collection from left to right? 
The `LET` clause can have more than one binding and in that case we want bindings to have access to their preceding `LET` bindings.

Reply:
I think because `seq` is a Kotlin [Sequence](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/-sequence/), which returns values through its iterator, the `map` will process the sequence from left to right.